### PR TITLE
Mention the Java liveheap subsampling in troubleshooting

### PR DESCRIPTION
**Pull request by jbachorik**
### What does this PR do? What is the motivation?
Adds a section about adjusting the subsampling rate for live heap profiling in the troubleshooting guide.

Related to [PROF-10303]

### Merge instructions

- [x] Please merge after reviewing

### Additional notes
This also subsumes changes from #18743 and makes the profile type format for Java compatible with other runtimes

[PROF-10303]: https://datadoghq.atlassian.net/browse/PROF-10303?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/profiler/profiler_troubleshooting/java.md
Comment (urseberry) [inline content/en/profiler/profiler_troubleshooting/java.md:108]:
```suggestion
If you are using the alpha feature of live heap profiling, you can tune the overhead by changing the percentage
```

**Review by urseberry — APPROVED** [review]

**Comment by urseberry** [comment]
Looks like the related code hasn't merged in yet? Ping us in #documentation in Slack when you are ready to merge this PR.

File: content/en/profiler/profiler_troubleshooting/java.md
Comment (jhgilbert) [inline content/en/profiler/profiler_troubleshooting/java.md:108]:
```suggestion
If you are using the alpha feature of live heap profiling, you can tune the overhead by changing the percentage
```
When I went to make this suggestion, I noticed this suggestion was added by a previous reviewer but wasn't applied -- this is the correct grammar, so I'm suggesting it again in case you accidentally resolved it last time. Otherwise, if you have objections, happy to chat about the logic behind it!

**Review by jhgilbert — APPROVED** [review]
Approved with an edit -- once that's applied, I can merge the PR, or happy to discuss further. Thanks!

**Pull request merged**
